src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -965,16 +950,40 @@ bool isRefEscape
             //•	the safe-to-escape of all argument expressions(including the receiver)
             //
 
+            if (symbol.IsStatic)
+            {
+                // ignore receiver when symbol is static
+                receiverOpt = null;
+            }
+
             //by default it is safe to escape
             uint escapeScope = Binder.ExternalScope;
 
             ArrayBuilder<bool> inParametersMatchedWithArgs = null;
 
-            if (!args.IsDefault)
+            if (!argsOpt.IsDefault)
             {
-                for (var argIndex = 0; argIndex < args.Length; argIndex++)
+moreArguments:
+                for (var argIndex = 0; argIndex < argsOpt.Length; argIndex++)
                 {
-                    RefKind effectiveRefKind = GetEffectiveRefKind(argIndex, argRefKinds, parameters, argsToParamsOpt, ref inParametersMatchedWithArgs);
+                    var argument = argsOpt[argIndex];
+                    if (argument.Kind == BoundKind.ArgListOperator)
+                    {
+                        Debug.Assert(argIndex == argsOpt.Length - 1, "vararg must be the last");
```

Comment:
Can you add a test trying to put vararg in the middle? Will the binding stop before reaching this point?

Reply:
I am pretty sure it will not be handles by binder, may not even parse. Varargs must be the last.

---
In reply to: [141156414](https://github.com/dotnet/roslyn/pull/22303#discussion_r141156414) [](ancestors = 141156414)

Reply:
CS0257ERR_VarargsLast      is a parser test that checks case where vararg is not last.

The assert here is mostly for informational purposes, also in case something changes in the future by accident, but that is unlikely.

---
In reply to: [141162613](https://github.com/dotnet/roslyn/pull/22303#discussion_r141162613) [](ancestors = 141162613,141156414)